{"create_time":"2025-04-05T09:05:02.852Z","id":"merged_4ffcbf49-9069-45d5-81c5-3dc73bf87e75","source_url":"file:///C:/Users/Dj/.rustup/toolchains/1.10-x86_64-pc-windows-msvc/share/doc/rust/html/std/io/enum.ErrorKind.html","merged_triples":[{"triples":[["std::io::ErrorKind","is_part_of","std::io"],["std::io::ErrorKind","description","A list specifying general categories of I/O error."],["std::io::ErrorKind::NotFound","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::NotFound","description","An entity was not found, often a file."],["std::io::ErrorKind::PermissionDenied","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::PermissionDenied","description","The operation lacked the necessary privileges to complete."],["std::io::ErrorKind::ConnectionRefused","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::ConnectionRefused","description","The connection was refused by the remote server."],["std::io::ErrorKind::ConnectionReset","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::ConnectionReset","description","The connection was reset by the remote server."],["std::io::ErrorKind::ConnectionAborted","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::ConnectionAborted","description","The connection was aborted (terminated) by the remote server."],["std::io::ErrorKind::NotConnected","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::NotConnected","description","The network operation failed because it was not connected yet."],["std::io::ErrorKind::AddrInUse","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::AddrInUse","description","A socket address could not be bound because the address is already in use elsewhere."],["std::io::ErrorKind::AddrNotAvailable","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::AddrNotAvailable","description","A nonexistent interface was requested or the requested address was not local."],["std::io::ErrorKind::BrokenPipe","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::BrokenPipe","description","The operation failed because a pipe was closed."],["std::io::ErrorKind::AlreadyExists","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::AlreadyExists","description","An entity already exists, often a file."],["std::io::ErrorKind::WouldBlock","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::WouldBlock","description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["std::io::ErrorKind::InvalidInput","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::InvalidInput","description","A parameter was incorrect."],["std::io::ErrorKind::InvalidData","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::InvalidData","description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["std::io::ErrorKind::TimedOut","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::TimedOut","description","The I/O operation's timeout expired, causing it to be canceled."],["std::io::ErrorKind::WriteZero","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::WriteZero","description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["std::io::ErrorKind::Interrupted","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::Interrupted","description","This operation was interrupted. Interrupted operations can typically be retried."],["std::io::ErrorKind::Other","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::Other","description","Any I/O error not part of this list."],["std::io::ErrorKind::UnexpectedEof","is_variant_of","std::io::ErrorKind"],["std::io::ErrorKind::UnexpectedEof","description","An error returned when an operation could not be completed because an end of file was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."],["std::io::ErrorKind","implements","std::fmt::Debug"],["std::io::ErrorKind","implements","std::clone::Clone"],["std::io::ErrorKind","implements","std::cmp::Eq"],["std::io::ErrorKind","implements","std::cmp::PartialEq"],["std::io::ErrorKind","implements","std::marker::Copy"],["std::io","contains","ErrorKind"],["ErrorKind","description","A list specifying general categories of I/O error."],["ErrorKind","since_version","1.0.0"],["ErrorKind","has_variant","NotFound"],["NotFound","description","An entity was not found, often a file."],["ErrorKind","has_variant","PermissionDenied"],["PermissionDenied","description","The operation lacked the necessary privileges to complete."],["ErrorKind","has_variant","ConnectionRefused"],["ConnectionRefused","description","The connection was refused by the remote server."],["ErrorKind","has_variant","ConnectionReset"],["ConnectionReset","description","The connection was reset by the remote server."],["ErrorKind","has_variant","ConnectionAborted"],["ConnectionAborted","description","The connection was aborted (terminated) by the remote server."],["ErrorKind","has_variant","NotConnected"],["NotConnected","description","The network operation failed because it was not connected yet."],["ErrorKind","has_variant","AddrInUse"],["AddrInUse","description","A socket address could not be bound because the address is already in use elsewhere."],["ErrorKind","has_variant","AddrNotAvailable"],["AddrNotAvailable","description","A nonexistent interface was requested or the requested address was not local."],["ErrorKind","has_variant","BrokenPipe"],["BrokenPipe","description","The operation failed because a pipe was closed."],["ErrorKind","has_variant","AlreadyExists"],["AlreadyExists","description","An entity already exists, often a file."],["ErrorKind","has_variant","WouldBlock"],["WouldBlock","description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["ErrorKind","has_variant","InvalidInput"],["InvalidInput","description","A parameter was incorrect."],["ErrorKind","has_variant","InvalidData"],["InvalidData","description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["InvalidData","since_version","1.2.0"],["ErrorKind","has_variant","TimedOut"],["TimedOut","description","The I/O operation's timeout expired, causing it to be canceled."],["ErrorKind","has_variant","WriteZero"],["WriteZero","description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["ErrorKind","has_variant","Interrupted"],["Interrupted","description","This operation was interrupted. Interrupted operations can typically be retried."],["ErrorKind","has_variant","Other"],["Other","description","Any I/O error not part of this list."],["ErrorKind","has_variant","UnexpectedEof"],["UnexpectedEof","description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."],["UnexpectedEof","since_version","1.6.0"],["ErrorKind","implements","std::fmt::Debug"],["ErrorKind","implements","std::clone::Clone"],["ErrorKind","implements","std::cmp::Eq"],["ErrorKind","implements","std::cmp::PartialEq"],["ErrorKind","implements","std::marker::Copy"],["std::io::ErrorKind","belongs_to_module","std::io"],["std::io::ErrorKind","has_description","A list specifying general categories of I/O error. This list is intended to grow over time and it is not recommended to exhaustively match against it."],["std::io::ErrorKind","since_version","1.0.0"],["std::io::ErrorKind","implements","Debug"],["std::io::ErrorKind","implements","Clone"],["std::io::ErrorKind","implements","Eq"],["std::io::ErrorKind","implements","PartialEq"],["std::io::ErrorKind","implements","Copy"],["std::io::ErrorKind","has_variant","NotFound"],["NotFound","has_description","An entity was not found, often a file."],["std::io::ErrorKind","has_variant","PermissionDenied"],["PermissionDenied","has_description","The operation lacked the necessary privileges to complete."],["std::io::ErrorKind","has_variant","ConnectionRefused"],["ConnectionRefused","has_description","The connection was refused by the remote server."],["std::io::ErrorKind","has_variant","ConnectionReset"],["ConnectionReset","has_description","The connection was reset by the remote server."],["std::io::ErrorKind","has_variant","ConnectionAborted"],["ConnectionAborted","has_description","The connection was aborted (terminated) by the remote server."],["std::io::ErrorKind","has_variant","NotConnected"],["NotConnected","has_description","The network operation failed because it was not connected yet."],["std::io::ErrorKind","has_variant","AddrInUse"],["AddrInUse","has_description","A socket address could not be bound because the address is already in use elsewhere."],["std::io::ErrorKind","has_variant","AddrNotAvailable"],["AddrNotAvailable","has_description","A nonexistent interface was requested or the requested address was not local."],["std::io::ErrorKind","has_variant","BrokenPipe"],["BrokenPipe","has_description","The operation failed because a pipe was closed."],["std::io::ErrorKind","has_variant","AlreadyExists"],["AlreadyExists","has_description","An entity already exists, often a file."],["std::io::ErrorKind","has_variant","WouldBlock"],["WouldBlock","has_description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["std::io::ErrorKind","has_variant","InvalidInput"],["InvalidInput","has_description","A parameter was incorrect."],["std::io::ErrorKind","has_variant","InvalidData"],["InvalidData","has_description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["std::io::ErrorKind","has_variant","TimedOut"],["TimedOut","has_description","The I/O operation's timeout expired, causing it to be canceled."],["std::io::ErrorKind","has_variant","WriteZero"],["WriteZero","has_description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["std::io::ErrorKind","has_variant","Interrupted"],["Interrupted","has_description","This operation was interrupted. Interrupted operations can typically be retried."],["std::io::ErrorKind","has_variant","Other"],["Other","has_description","Any I/O error not part of this list."],["std::io::ErrorKind","has_variant","UnexpectedEof"],["UnexpectedEof","has_description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."]],"startIndex":0,"endIndex":39000}],"changes_info_triples":[],"model":"deepseek-r1-250120"}